packages/flutter_tools/lib/src/ios/migrations/storyboard_version_migration.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/logger.dart';
+import '../../base/project_migrator.dart';
+import '../../xcode_project.dart';
+
+// Update Main.storyboard tools versions to avoid Xcode UI hang.
+class StoryboardVersionMigration extends ProjectMigrator {
+  StoryboardVersionMigration(
+    IosProject project,
+    Logger logger,
+  ) : _mainStoryboard = project.xcodeMainStoryboard,
+      super(logger);
+
+  final File _mainStoryboard;
+
+  @override
+  bool migrate() {
+    if (_mainStoryboard.existsSync()) {
+      processFileLines(_mainStoryboard);
+    } else {
+      logger.printTrace('Xcode project main storyboard not found, skipping version migration.');
+    }
+
+    return true;
+  }
+
+  @override
+  String migrateLine(String line) {
+    String updatedString = line;
+    final List<String> originalDocumentHeaders = <String>[
+      // https://github.com/flutter/flutter/commit/5f6e9cb39c
+      '<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="9531"',
+      // https://github.com/flutter/flutter/pull/4252
+      '<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="8150"',
+      // https://github.com/flutter/flutter/pull/4277
+      '<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="10117"',
+      // https://github.com/flutter/flutter/pull/4820
+      '<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="6211"',
+      // https://github.com/flutter/flutter/pull/11505
+      '<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="12121"',
+    ];
+
+    for (final String original in originalDocumentHeaders) {
+      if (!line.contains(original)) {
+        continue;
+      }
+      const String mainReplacement = '<document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="13122.16" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES" initialViewController="BYZ-38-t0r">';
+      updatedString = mainReplacement;
+      if (!migrationRequired) {
+        // Only print once.
+        logger.printStatus('Updating Xcode storyboard versions.');
+      }
```

Comment:
I didn't get this. Why we log `Updating Xcode storyboard versions.` when "migration is not required"

Reply:
Ok, can we add more comments to explain it. (I think just copy paste your reply here is fine too)
For someone like me who is not familar with a `migration`, this is a little confusing (even with the line `// Only print once.`)

Reply:
Actually I can just remove the `if`, there's only one `document` tag in a storyboard.
The original version of this change also migrated `LaunchScreen.storyboard`, hence the loop and making sure this is only printed once.  But `LaunchScreen.storyboard` doesn't make Xcode SPOD so I removed it to minimize the migration.